src/app/user-event/assessment-event.component.ts

```diff
@@ -299,65 +291,165 @@ export class AssessmentEventComponent extends ListComponentDirective<AssessmentM
+    this._surveys = !!assessmentModelWithoutUser ? assessmentModelWithoutUser.forms : null;
 
-          this._isAnswered = !list.find((item: AssessmentModel) => !item.isAnswered);
+    this._users.sort((x, y) => x.user.name < y.user.name ? -1 : 1);
 
-          if (this._assessmentObject && this._assessmentObject.userId) {
-            this._assessmentObject = list.find((item: AssessmentModel) => item.user.id === this._assessmentObject.userId);
-          }
-        })
+    this._isAnswered = !list.some((item: AssessmentModel) => !item.isAnswered);
+
+    if (this._assessmentObject instanceof AssessmentModel && this._assessmentObject.user) {
+      this._assessmentObject = list.find(
+        (item: AssessmentModel) => item.user.id === (<AssessmentModel>this._assessmentObject).user.id
       );
+    }
   }
 
-  protected _fetch(): Observable<any> {
-    let observable = new Observable((observer) => {
-      this._fetching = this._service.list(this._queryParams).subscribe((res: IListResponse<AssessmentModel>) => {
-        let list = res.data;
+  protected _fetch(loadLinkedData: boolean): Observable<any> {
+    return this._service.list(this._queryParams)
+      .pipe(
+        map((res: IListResponse<AssessmentModel>) => {
+          const items = res.data;
+          this._meta = res.meta;
 
-        list.forEach((assessment) => {
-          assessment.isClassic = !!assessment.user;
-          assessment.isAnswered = !assessment.forms.find(x => x.status === AssessmentFormStatus.New);
-          assessment.forms
-            .sort((x: IFormAnswer, y: IFormAnswer) => x.form.name < y.form.name ? -1 : 1)
-            .forEach(form => form.active = false);
-          if (assessment.user && assessment.user.hasPicture) {
-            this._userPictureService.getPicture(assessment.user.id).subscribe(pic => assessment.user.picture = pic);
+          for (const item of items) {
+            item.setFormsActive(false);
           }
 
-          this._isClear = !assessment.isAnswered;
-        });
+          if (this._project.isLast) {
+            let forms = items[items.length - 1].forms;
+            forms[forms.length - 1].isLast = true;
+          }
 
-        if (this._project.isLast) {
-          let forms = list[list.length - 1].forms;
-          forms[forms.length - 1].isLast = true;
-        }
+          return items;
+        }),
+        switchMap((items: AssessmentModel[]) => from(items).pipe(
+          mergeMap((item: AssessmentModel) => loadLinkedData
+            ? this._fetchItemLinkedData(item)
+            : this._getItemLinkedDataFromCache(item),
+            this._maxConcurrency
+          ),
+          toArray()
+        ))
+      );
+  }
 
-        observer.next(list);
-        observer.complete();
-      }, error => observer.error(error));
-    });
-    return observable;
+  protected _fetchItemLinkedData(item: AssessmentModel): Observable<AssessmentModel> {
+    if (!item.user || !item.user.hasPicture) {
```

Comment:
Is property `hasPicture` valid in the system? There is no code to set this property. Can it be replaced by `user.picture`?

Reply:
Sure, property `hasPicture` is a valid property from backend response, it needs to indicate is user has a picture or not, since picture load by separate resquest.